### PR TITLE
Hmedina patch 2

### LINKED ISSUE #650: OCaml 5: explicit unix library loading
OCaml 5 seems to have changed library loading, and the unix standard tool-chain needs an explicit invocation.

```
#=== ERROR while compiling kappa-binaries.4.1~dev =============================#
# context     2.1.2 | linux/x86_64 | ocaml.5.0.0 | pinned(git+https://github.com/Kappa-Dev/KappaTools.git#b4abe7ec)
# path        ~/.opam/default/.opam-switch/build/kappa-binaries.4.1~dev
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p kappa-binaries -j 11
# exit-code   1
# env-file    ~/.opam/log/kappa-binaries-2127-0a6acd.env
# output-file ~/.opam/log/kappa-binaries-2127-0a6acd.out
### output ###
# [...]
# (cd _build/default && /home/hmedina/.opam/default/bin/ocamlopt.opt -w -40 -open Kappa_grammar -open Kappa_terms -open Kappa_mixtures -open Kappa_parameters -open Kappa_generic_toolset -open Kappa_logging -open Kappa_cli -open Kappa_runtime -open Kappa_kasa_type_interface -g -I core/symmetries/.kappa_symmetries.objs/byte -I core/symmetries/.kappa_symmetries.objs/native -I /home/hmedina/.opam/d[...]
# File "_none_", line 1:
# Alert ocaml_deprecated_auto_include:
# OCaml's lib directory layout changed in 5.0. The unix subdirectory has been
# automatically added to the search path, but you should add -I +unix to the
# command-line to silence this alert (e.g. by adding unix to the list of
# libraries in your dune file, or adding use_unix to your _tags file for
# ocamlbuild, or using -package unix for ocamlfind).
# (cd _build/default && /home/hmedina/.opam/default/bin/ocamlopt.opt -w -40 -open Kappa_generic_toolset -open Kappa_mixtures -open Kappa_terms -open Kappa_runtime -open Kappa_grammar -open Kappa_cli -open Kappa_logging -open Kappa_parameters -open Kappa_staticanalyses -open Kappa_symmetries -open Kappa_reachability -open Kappa_kasa_export -open Kappa_kasa_type_interface -g -o core/odes/KaDE.exe[...]
# File "_none_", line 1:
# Error: No implementations provided for the following modules:
#          Unix referenced from core/cli/kappa_cli.cmxa(Kappa_cli__Kappa_files)

```

### PR DESCRIPTION
**Pull request by hmedina**
Fix #650 by explicitly stating the unix library imports. Most ka-things import kappa_generic_toolset, and so adding `unix` to that library's Dune stanza solved most of the warnings. The version object got a command-line flag instead.

Changes to the continuous integration script are just to strengthen it against a known failure mode, as documented in that commit message (https://github.com/Kappa-Dev/KappaTools/commit/d796fa7e39c918bd4a6dc881d0c2b6efb7ef0ccd)

Although the KappaTools compile under OCaml 5, the documentation appears to require `tyxml`, which does not yet support OCaml 5 (see https://github.com/Kappa-Dev/KappaTools/commit/b9fe862fea9718eada0edb2810328cc63ddacfcf), and so OCaml 5 can't be included in the build matrix (yet...)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,19 +9,19 @@ jobs:
         ocaml-compiler:
           - 4.13.x
           - 4.14.x
+          - 5.0.x
         additional-packages:
-          -
```
Comment (pirbo) [inline .github/workflows/ci.yml:13]: This line is actually important. It expresses that we should try to compile with **and without** labltk installed...

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -78,13 +78,13 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+      - run: sudo apt-get update
       - name: Setup OCaml
         uses: ocaml/setup-ocaml@v2
         with:
           ocaml-compiler: 4.13.x
```
Comment (pirbo) [inline .github/workflows/ci.yml:84]: Can be perfectly done in an other PR but reviewing this makes me realize that `4.14`should be the "default" version used to build the js app and under MacOS/Windows...

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -9,19 +9,19 @@ jobs:
         ocaml-compiler:
           - 4.13.x
           - 4.14.x
+          - 5.0.x
```
Comment (pirbo) [inline .github/workflows/ci.yml]: Making kappa binaries compile with OCaml 5 is a mandatory step that has to be done and is well done here but I fear that sadly, the journey to make KaSim "compatible" with OCaml5(while keeping it compatible with OCaml 4) is more complex than that! 
I'll confirm my fears tomorrow (it's as simple as running `make check`) but OCaml 5 changes the random number generator provided by the standard library (to make it multithread capable) so `--seed 42` does not gives the same simulation when compiled with Ocaml 4 and OCaml 5 :-(

Comment (hmedina) [reply .github/workflows/ci.yml]: I can confirm essentially the entire test suite fails when `make check`'ed under OCaml 5

Comment (feret) [reply .github/workflows/ci.yml]: Testing the test suite on OCaml 4 and OCaml 5 with the same reference files does not make sense. 
The random generators are not the same, so most stochastic parts will differ.

**Comment by antoinepouille** [comment]
Closing as it seems not to be relevant anymore?

**Pull request closed without merging**